lib/passport/controllers/two_factor_auth_controller.ex

```diff
@@ -0,0 +1,126 @@
+defmodule Passport.TwoFactorAuthController do
+  defmacro __using__(opts) do
+    quote location: :keep do
+      @behaviour Passport.TwoFactorAuthController
+
+      require Passport.Config
+
+      if Keyword.has_key?(unquote(opts), :two_factor_auth_model) do
+        @impl true
+        def two_factor_auth_model, do: unquote(opts[:two_factor_auth_model])
+      end
+
+      @doc """
+      POST /account/confirm/tfa
+      """
+      def create(conn, params) do
+        Passport.TwoFactorAuthController.create(__MODULE__, conn, params)
+      end
+
+      @doc """
+      GET /account/confirm/tfa/:token
+      """
+      def show(conn, params) do
+        Passport.TwoFactorAuthController.show(__MODULE__, conn, params)
+      end
+
+      @doc """
+      POST /account/confirm/tfa/:token
+      """
+      def confirm(conn, params) do
+        Passport.TwoFactorAuthController.confirm(__MODULE__, conn, params)
+      end
+
+      @doc """
+      DELETE /account/confirm/tfa/:token
+      """
+      def delete(conn, params) do
+        Passport.TwoFactorAuthController.delete(__MODULE__, conn, params)
+      end
+
+      defoverridable [create: 2, show: 2, confirm: 2, delete: 2]
+    end
+  end
+
+  import Plug.Conn
+  import Phoenix.Controller
+  import Passport.APIHelper
+
+  @doc """
+  two_factor_auth_model denotes what module represents the TwoFactorAuth entity.
+
+  This function should return a valid Ecto.Schema model
+  """
+  @callback two_factor_auth_model() :: atom
+
+  def create(_controller, conn, params) do
+    case Passport.Sessions.authenticate_entity(params["email"], params["password"]) do
+      {:ok, %{tfa_enabled: true} = entity} ->
+        send_unauthorized(conn, reason: "TFA already enabled")
```

Comment:
Not sure if "already done" should return an unauthorized error

Reply:
409 Conflict maybe